payments-core/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethod.kt

```diff
@@ -0,0 +1,36 @@
+package com.stripe.android.paymentsheet.model
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import com.stripe.android.R
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Enum defining all payment methods supported in Payment Sheet.
+ */
+internal enum class SupportedPaymentMethod(
```

Comment:
I think in time we will see what type of interface makes sense here.  Enums are nice when you need to switch on it to perform different operations for each, but ideally nothing outside this class would need to do special processing for the different payment methods.

It also feels like that this might conflict slightly with the open close principle in that it will often need modifications for each new class added.

Might be better to talk through synchronously.